cmds/switch_root/utils.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+const AT_REMOVEDIR = 0x200
+
+// Code in this file comes from the syscall package.
+// Either the things are not out yet (fstatat) or private (unlinkat with flags)
+//
+// ulinkat because it's private
+// fstatat because it's commited but not released yet
+
+// Do the interface allocations only once for common
+// Errno values.
+var (
+	errEAGAIN error = syscall.EAGAIN
+	errEINVAL error = syscall.EINVAL
+	errENOENT error = syscall.ENOENT
+)
+
+// TODO remove when available/public in go runtime
```

Comment:
This seems like a premature optimization. The allocation incurs about 50-100 nanoseconds depending on what you do with the error (compare errors, switch on error, or return, etc). Do you really care about that kind of performance here? If you do, I'd like to see a benchmark attached to the PR.

Reply:
Yeah, this is more copy/paste from the golang runtime, I hope to get rid of this with the newer syscall package.